src/frontend/src/scheduler/distributed/stage.rs

```diff
@@ -183,14 +181,15 @@ impl StageExecution {
     /// Starts execution of this stage, returns error if already started.
     pub async fn start(&self) {
         let mut s = self.state.write().await;
-        match &*s {
-            &StageState::Pending => {
+        let cur_state = mem::replace(&mut *s, StageState::Failed);
```

Comment:
Why should we replace the state with `Failed`? Even though it won't cause any bugs, it looks counter-intuitive. Same as 

https://github.com/risingwavelabs/risingwave/blob/b5f84f32e0f20b59f161bef803bbb1a4dfd51b8a/src/frontend/src/scheduler/distributed/query.rs#L163

Here, `Failed` is just a dummy value.

Reply:
cc. @liurenjie1024

Reply:
I think this is try to avoid rust compiler complain on lifetime and ownership?

Reply:
Anyway, feel free to submit PR to fix this if you have better impl with the same semantics 😆